### DISCUSSION TITLE
Help me understand the parallelization?

### DISCUSSION THREAD

**Discussion by rohit1998** [optuna/optuna#4812] [Q&A]
I am using this code to run my study on a single node with 4 cores.

```
def objective(config):
    # some objective

def my_app(config):
    study = optuna.create_study(
        study_name="my_study", storage="mysql+pymysql://user@localhost/my_study", load_if_exists=True, direction='maximize',

    )
    func = lambda trial: objective(trial, config)
    study.optimize(func, n_trials=100)
```

i did two experiments. I ran my script 4 times in different terminal and got 4x parallelization, expecting a total of 400 trials, 100 each. But my memory consumption after 10 trials (per process) became a lot.

So I reduced my `n_trials = 10`. And used bash to call my script 40 times, such that only 4 run at a time, again achieving that 4x parallelization. This time memory consumption was low because each python process ran 10 trials than exited freeing up all the memory. My study finished successfully, all 400 trials.

I have two questions now
1. Will second scenario be as efficient as 1 in terms of convergence. In particular is running 4 study with 100 trials or 40 study with 10 trials similar. I am using default TPE sampler.
2. Is there a memory leakage in my objective function? I did try dask parallelization and putting `gc_after_trial = True`  but didnt see any improvement in memory consumption.

Thanks.

**Comment by nzw0301** [comment]
1. In terms of the behaviour of `TPESampler`, I suppose so. 
2. I think so; it depends on the script.
